### PR TITLE
Refactor the guardio sample application to add eslint file

### PR DESCRIPTION
**Pull request by Achintha444**

### CODE REVIEW

**Review by wenodpathirana — APPROVED** [review]

File: b2b-sample/guardio-insurance-sample-application/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+         ],
+         "tsdoc/syntax": "warn"
+     },
+     settings: {
+         react: {
+             version: "detect"
+         }
+     }
+ };
+
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/.eslintrc.js:331]: Add a new line

File: b2b-sample/guardio-insurance-sample-application/.eslintrc.json [hunk 2]
```diff
@@ -1,24 +1,26 @@
...
+                "max": 1,
+                "maxEOF": 1
+            }
+        ],
+        "eol-last": [
+            "error",
+            "always"
+        ]
+    }
+}
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/.eslintrc.json:26]: Add a new line

File: b2b-sample/guardio-insurance-sample-application/components/common/dialog.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+ * under the License.
+ */
+
+import React from "react";
+import { Notification } from "rsuite";
+
+/**
+ * 
+ * @param prop - type (error, info, success ), header - title text, body - body text
+ * @returns A side dialog to show notifications
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/components/common/dialog.js:25]:
```suggestion
 * @param prop - type (error, info, success ), header - title text, body - body text
 *
 * @returns A side dialog to show notifications
```

File: b2b-sample/guardio-insurance-sample-application/components/common/logo/logoComponent.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+import React from "react";
+import Logo from "./logo";
+import config from "../../../config.json";
+import styles from "../../../styles/Settings.module.css";
+
+/**
+ * 
+ * @param prop - name (org name), imageSize (small, medium, large, x-large)
+ * @returns
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/components/common/logo/logoComponent.js:27]:
```suggestion
 * @param prop - name (org name), imageSize (small, medium, large, x-large)
 *
 * @returns 
```

File: b2b-sample/guardio-insurance-sample-application/components/sections/dashboardSection/dashboardSectionComponent.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+import LatestNewsComponent from "./otherComponents/latestNewsComponent";
+import UserDetails from "./otherComponents/userDetails";
+import styles from "../../../styles/Settings.module.css";
+import decodeMe from "../../../util/apiDecode/dashboard/decodeMe";
+import LogoComponent from "../../common/logo/logoComponent";
+
+/**
+ * 
+ * @param prop - session, orgName
+ * @returns Dashboard interface section
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/components/sections/dashboardSection/dashboardSectionComponent.js:31]:
```suggestion
 * @param prop - session, orgName
 *
 * @returns Dashboard interface section
```

File: b2b-sample/guardio-insurance-sample-application/components/sections/dashboardSection/otherComponents/newsComponent.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+import image1 from "../../../../public/internal/news1.jpeg";
+import image2 from "../../../../public/internal/news2.jpeg";
+import image3 from "../../../../public/internal/news3.jpeg";
+import image4 from "../../../../public/internal/news4.jpeg";
+import { getCurrentDate } from "../../../../util/util/common/common";
+
+/**
+ * 
+ * @param prop - header (header text), body (body text)
+ * @returns Single news component
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/components/sections/dashboardSection/otherComponents/newsComponent.js:31]:
```suggestion
 * @param prop - header (header text), body (body text)
 *
 * @returns Single news component
```

File: b2b-sample/guardio-insurance-sample-application/components/sections/dashboardSection/otherComponents/userDetails.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+import Image from "next/image";
+import React from "react";
+import { Panel } from "rsuite";
+import profileImage from "../../../../public/internal/profile.svg";
+import styles from "../../../../styles/Settings.module.css";
+
+/**
+ * 
+ * @param prop - me (details of the logged in user)
+ * @returns The profile details section
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/components/sections/dashboardSection/otherComponents/userDetails.js:28]:
```suggestion
 * @param prop - me (details of the logged in user)
 *
 * @returns The profile details section
```

File: b2b-sample/guardio-insurance-sample-application/components/sections/home.js [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+import Custom500 from "../../pages/500";
+import styles from "../../styles/Settings.module.css";
+import { checkCustomization, hideBasedOnScopes } from "../../util/util/frontendUtil/frontendUtil";
+import { orgSignout } from "../../util/util/routerUtil/routerUtil";
+import LogoComponent from "../common/logo/logoComponent";
+
+/**
+ * 
+ * @param prop - orgId, name, session, colorTheme
+ * @returns The home section. Mainly side nav bar and the section to show other settings sections.
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/components/sections/home.js:36]:
```suggestion
 * @param prop - orgId, name, session, colorTheme
 *
 * @returns The home section. Mainly side nav bar and the section to show other settings sections.
```

File: b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/data/templates/enterprise-identity-provider.json [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+        },
+        "templateId": "enterprise-idp"
+    },
+    "image": "enterprise",
+    "name": "Enterprise",
+    "services": [],
+    "disabled": false,
+    "templateId": "enterprise-idp",
+    "type": "ENTERPRISE"
+}
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/data/templates/enterprise-identity-provider.json:142]: Add a new line

File: b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/data/templates/google.json [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+            },
+            "outboundConnectors": {
+                "connectors": []
+            }
+        },
+        "templateId": "google-idp"
+    },
+    "type": "SOCIAL",
+    "templateId": "google-idp"
+}
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/data/templates/google.json:116]: Add a new line

File: b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/otherComponents/buttonGroupIdentityProviderDetails.js [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+    const onAddToLoginFlowClick = () => {
+        setOpenListAppicationModal(true);
+    };
+
+    const onCloseListAllApplicaitonModal = () => {
+        setOpenListAppicationModal(false);
+    };
+
+    return (
+
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/otherComponents/buttonGroupIdentityProviderDetails.js:101]:
```suggestion
```

File: b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/otherComponents/identityProviderList.js [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+
+import React from "react";
+import { PanelGroup } from "rsuite";
+import IdentityProviderDetails from "./identityProviderDetails";
+import styles from "../../../../../styles/idp.module.css";
+
+/**
+ * 
+ * @param prop - idpDetails (List of idp's), fetchAllIdPs (function to fetch all idp's), session
+ * @returns List of idp's created in the organization
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/otherComponents/identityProviderList.js:27]:
```suggestion
 * @param prop - idpDetails (List of idp's), fetchAllIdPs (function to fetch all idp's), session
 *
 * @returns List of idp's created in the organization
```

File: b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/otherComponents/idpDetailsSections/general.js [hunk 13]
```diff
@@ -0,0 +1,168 @@
...
+    "../../../../../../util/apiDecode/settings/identityProvider/decodePatchGeneralSettingsIdp";
+import { checkIfJSONisEmpty } from "../../../../../../util/util/common/common";
+import { LOADING_DISPLAY_BLOCK, LOADING_DISPLAY_NONE } from "../../../../../../util/util/frontendUtil/frontendUtil";
+import { errorTypeDialog, successTypeDialog } from "../../../../../common/dialog";
+import HelperText from "../../../../../common/helperText";
+
+/**
+ * 
+ * @param prop - fetchData (function to fetch data after form is submitted), session, idpDetails
+ * @returns The general section of an idp
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/otherComponents/idpDetailsSections/general.js:34]:
```suggestion
 * @param prop - fetchData (function to fetch data after form is submitted), session, idpDetails
 * 
 * @returns The general section of an idp
```

File: b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/otherComponents/idpDetailsSections/settings.js [hunk 14]
```diff
@@ -0,0 +1,137 @@
...
+import decodeUpdateFederatedAuthenticators from
+    "../../../../../../util/apiDecode/settings/identityProvider/decodeUpdateFederatedAuthenticators";
+import { checkIfJSONisEmpty } from "../../../../../../util/util/common/common";
+import { LOADING_DISPLAY_BLOCK, LOADING_DISPLAY_NONE } from "../../../../../../util/util/frontendUtil/frontendUtil";
+import { errorTypeDialog, successTypeDialog } from "../../../../../common/dialog";
+
+/**
+ * 
+ * @param prop - session, idpDetails
+ * @returns The settings section of an idp
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/otherComponents/idpDetailsSections/settings.js:36]:
```suggestion
 * @param prop - session, idpDetails
 *
 * @returns The settings section of an idp
```

File: b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/otherComponents/idpDetailsSections/settingsFormSection/settingsFormSelection.js [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+        return keyFederatedAuthenticator ? keyFederatedAuthenticator.value : "";
+    };
+
+    const copyValueToClipboard = (text) => {
+        copyTheTextToClipboard(text);
+        infoTypeDialog(toaster, "Text copied to clipboard");
+    };
+
+    return (
+
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/otherComponents/idpDetailsSections/settingsFormSection/settingsFormSelection.js:63]:
```suggestion
```

File: b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/otherComponents/idpDetailsSections/settingsFormSection/settingsFormSelection.js [hunk 16]
```diff
@@ -0,0 +1,107 @@
...
+                                    { meta.error }
+                                </FormSuite.ErrorMessage>) }
+
+                            </FormSuite.Group>
+                        ) }
+                    />
+                );
+            })
+            : <p>Access the console to edit this identity provider</p>
+
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/otherComponents/idpDetailsSections/settingsFormSection/settingsFormSelection.js:104]:
```suggestion
```

**Review by thivi — APPROVED** [review]

**Review by thivi — APPROVED** [review]

File: b2b-sample/guardio-insurance-sample-application/components/common/logo/logoComponent.js [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+
+import React from "react";
+import Logo from "./logo";
+import config from "../../../config.json";
+import styles from "../../../styles/Settings.module.css";
+
+/**
+ * 
+ * @param prop - name (org name), imageSize (small, medium, large, x-large)
+ *
```
Comment (sandushi) [inline b2b-sample/guardio-insurance-sample-application/components/common/logo/logoComponent.js:27]: Do we need this new line in this comment? some places have, some places don't have this. Please follow one manner.

File: b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/data/templates/enterprise-identity-provider.json [hunk 18]
```diff
@@ -0,0 +1,142 @@
...
+                            "isMandatory": true,
+                            "isConfidential": false,
+                            "options": [],
+                            "defaultValue": "",
+                            "subProperties": []
+                        },
+                        {
+                            "key": "callbackUrl",
+                            "displayName": "Callback Url",
+                            "description": "The WSO2 Identity Server URL to which the user needs to be redirected after completing the authentication at the identity provider. The identity provider needs to send the authorization code to this URL upon successful authentication.",
```
Comment (sandushi) [inline b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/data/templates/enterprise-identity-provider.json:84]: Please check this line doesn't exceed the line length

File: b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/idpSectionComponent.js [hunk 19]
```diff
@@ -0,0 +1,446 @@
...
+            setSelectedTemplate(undefined);
+        } else {
+            errorTypeDialog(toaster, "Error Occured", "Error occured while creating the identity provider. Try again.");
+        }
+    };
+
+    const onIdPSave = async (formValues, template) => {
+
+        if (checkEmpty(formValues, template)) {
+            errorTypeDialog(toaster, "Fields Cannot be Empty", "Form fields cannot be empty occured.");
```
Comment (sandushi) [inline b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/idpSectionComponent.js:102]: `Fields cannot be empty"`

File: b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/idpSectionComponent.js [hunk 20]
```diff
@@ -0,0 +1,446 @@
...
+        } else {
+            decodeCreateIdentityProvider(session, template, formValues)
+                .then((response) => onIdpCreated(response));
+        }
+
```
Comment (sandushi) [inline b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/idpSectionComponent.js:107]: Remove this new line

File: b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/idpSectionComponent.js [hunk 21]
```diff
@@ -0,0 +1,446 @@
...
+ *              (what will happen when a particular template is selected)
+ * @returns A modal to select idp's
+ */
+const AddIdentityProviderModal = (prop) => {
+
+    const { openModal, onClose, templates, onTemplateSelected } = prop;
+
+    const resolveIconName = (template) => {
+        if (GOOGLE_ID === template.templateId) {
+
```
Comment (sandushi) [inline b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/idpSectionComponent.js:226]: Shall we check that this new line is required after if statement? Check other places as well.

File: b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/otherComponents/buttonGroupIdentityProviderDetails.js [hunk 22]
```diff
@@ -0,0 +1,127 @@
...
+    useEffect(() => {
+        if (!checkIfJSONisEmpty(applicationDetail)) {
+            setIdpIsinAuthSequence(checkIfIdpIsinAuthSequence(applicationDetail, idpDetails));
+        }
+    }, [ idpDetails, applicationDetail ]);
+
+    const onIdpDelete = (response) => {
+        if (response) {
+            successTypeDialog(toaster, "Success", "Identity Provider Deleted Successfully");
```
Comment (sandushi) [inline b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/otherComponents/buttonGroupIdentityProviderDetails.js:78]: `Identity Provider is deleted successfully`

File: b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/manageUserSection/otherComponents/editUserComponent.js [hunk 23]
```diff
@@ -0,0 +1,221 @@
...
+    const firstNameValidate = (firstName, errors) => {
+        if (!firstName) {
+            errors.firstName = "This field cannot be empty";
+        }
+
+        return errors;
+    };
+
+    const familyNameValidate = (familyName, errors) => {
+        if (!familyName) {
```
Comment (sandushi) [inline b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/manageUserSection/otherComponents/editUserComponent.js:52]: Shall we use one common function to validate these fields other that implementing method for each validation?

File: b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/otherComponents/buttonGroupIdentityProviderDetails.js (hunk 22 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/otherComponents/buttonGroupIdentityProviderDetails.js:78]: Fixed with https://github.com/wso2/samples-is/pull/416/commits/130d957728499e5e0e731eba0b663086f931fe95

File: b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/data/templates/enterprise-identity-provider.json (hunk 18 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/data/templates/enterprise-identity-provider.json:84]: This is a template file, and IMO it is okay to exceed 120 characters in such files, as we have done the same in our other repos as well.

ex: 
- https://github.com/wso2-enterprise/asgardeo-apps/blob/58c7f91640fae55d4215c33a7075399aeabe7fe2/apps/console/src/extensions/identity-provider-templates/templates/saml/saml.json#L61

File: b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/idpSectionComponent.js (hunk 21 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/idpSectionComponent.js:226]: Fixed with https://github.com/wso2/samples-is/pull/416/commits/130d957728499e5e0e731eba0b663086f931fe95

File: b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/manageUserSection/otherComponents/editUserComponent.js (hunk 23 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/manageUserSection/otherComponents/editUserComponent.js:52]: This is fixed with a later PR,

### File
- https://github.com/wso2/samples-is/blob/master/b2b-sample/libs/shared/util/util-front-end-util/src/lib/formUtil/formUtil.ts 
- https://github.com/wso2/samples-is/blob/09d8f7d6abe21d199f2c739e578b0883825d8cf2/b2b-sample/apps/business-admin-app/components/sections/sections/settingsSection/manageUserSection/otherComponents/editUserComponent.tsx#L110

### Commit
- https://github.com/wso2/samples-is/commit/9c330e205151b20f9436bf6aad9c4db85e216673#diff-2982d16cf307c73b8cbce49859f3040b0fa42f10c49acaa3952642cdd40632b5

File: b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/idpSectionComponent.js (hunk 19 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/idpSectionComponent.js:102]: This was fixed when this file was moved to a different location

### File
- https://github.com/wso2/samples-is/blob/master/b2b-sample/apps/business-admin-app/components/sections/sections/settingsSection/idpSection/idpSectionComponent.tsx

File: b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/idpSectionComponent.js (hunk 20 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/components/sections/settingsSection/idpSection/idpSectionComponent.js:107]: This was fixed when this file was moved to a different location

### File
- https://github.com/wso2/samples-is/blob/master/b2b-sample/apps/business-admin-app/components/sections/sections/settingsSection/idpSection/idpSectionComponent.tsx

**Pull request merged**